### DISCUSSION TITLE
How to get available glyphs and their name

### DISCUSSION THREAD

**Discussion by workgroupengineering** [SixLabors/Fonts#343] [Q&A] [answered]
I load font like this:

```csharp
var fonts = new FontCollection();

var fontFamily = fonts.Add("myfont.ttf");
var font = fontFamily.CreateFont(12);
```
Now, how to get available glyphs and their name?

**Comment by JimBobSquarePants** [comment]
We don't actually decode the names of glyphs as many fonts don't store them and it would massivly bloat the memory used.

It is possible to get all the glyphs, metrics, and codepoints though. 

First use ` Font.FontMetrics.GetAvailableCodePoints();` to get all the codepoints.

Then loop through and use `Font.TryGetGlyphs(...)` to get individual glyphs.

**Reply by workgroupengineering** [reply]
Thanks for the reply. I already tried with this code. But I couldn't get the name of glyph.

```csharp
var codepoints = font.FontMetrics.GetAvailableCodePoints();

foreach (var codepoint in codepoints)
{

    if (font.TryGetGlyphs(codepoint, out var glyphs) && glyphs.Count > 0)
    {
        
        foreach (var glyph in glyphs.Where(g => g.GlyphMetrics.GlyphType != GlyphType.Fallback))
        {
            var glyphName = string.Empty;
            // How get glyphName
            Console.WriteLine($"Codeponint U{codepoint.Value:X} -> {glyphName}");
        }
    }
}
```

**Reply by JimBobSquarePants** [reply]
I’ve just told you that we don’t decode the names and also why we don’t do so.

**Reply by workgroupengineering** [reply]
I understand. I was just hoping you could give me some pointers on how to get them. I fully understand the need to optimize memory usage. If you give me some indications where to look I would like to try to implement a method extension to get the name of the glyph.

**Reply by JimBobSquarePants** [reply, marked as answer]
We skip it entirely. There's no way for you to get it using this library.
